### PR TITLE
feat: Use reshapes and pad to support slice with steps

### PR DESCRIPTION
**Pull request by p-lanza**
The comments should explain how the conversion works, but do let me know if something is not clear.

### CODE REVIEW

File: src/Conversion/ONNXToTOSA/Tensor/Slice.cpp
Comment (jorickert) [inline src/Conversion/ONNXToTOSA/Tensor/Slice.cpp:1]: I think we lost the check for negative steps. Could you add a test for them?

Comment (p-lanza) [reply src/Conversion/ONNXToTOSA/Tensor/Slice.cpp:1]: Shape inference had a bug for negative steps. Fixed it but failing the conversion if we see negative steps. I will leave supporting negative steps for future work.

File: test/mlir/conversion/onnx_to_tosa/Tensor/Slice.mlir [hunk 1]
```diff
@@ -57,3 +57,88 @@ func.func @slice_default_axes(%arg0: tensor<20x10x5xf32>,
...
+  %starts = "onnx.Constant"() {value = dense<[5, 10]> : tensor<2xi64> } : () -> tensor<2xi64>
+  %ends = "onnx.Constant"() {value = dense<[97, 192]> : tensor<2xi64> } : () -> tensor<2xi64>
+  %steps = "onnx.Constant"() {value = dense<[5, 10]> : tensor<2xi64> } : () -> tensor<2xi64>
+  %1 = "onnx.Slice"(%arg0, %starts, %ends, %axes, %steps) : (tensor<99x195xf32>, tensor<2xi64>, tensor<2xi64>, tensor<2xi64>, tensor<2xi64>) -> tensor<19x19xf32>
+  return %1 : tensor<19x19xf32> 
+}
+// CHECK-LABEL: func @slice_steps_and_edges_with_padding
+// CHECK: %0 = "tosa.const"() <{value = dense<[0, 0, 1, 5]> : tensor<4xi64>}> : () -> tensor<4xi64>
+// CHECK: %1 = "tosa.const"() <{value = dense<0.000000e+00> : tensor<f32>}> : () -> tensor<f32>
+// CHECK: %2 = tosa.pad %arg0, %0, %1 : (tensor<99x195xf32>, tensor<4xi64>, tensor<f32>) -> tensor<100x200xf32>
```
Comment (jorickert) [inline test/mlir/conversion/onnx_to_tosa/Tensor/Slice.mlir]: you will need to update this pad to take a tosa.const_shape

**Review by jorickert — APPROVED** [review]
Nice implementation, they diagrams also help a lot 💯

File: test/mlir/conversion/onnx_to_tosa/Tensor/Slice.mlir
Comment (jorickert) [inline test/mlir/conversion/onnx_to_tosa/Tensor/Slice.mlir:1]: (Assuming they do not exist yet), could you add some tests for edge cases?
- Step > dim size
- Start > dim size
